src/main/java/com/avispl/symphony/dal/communicator/aver/ptz/AverPTZCommunicator.java

```diff
@@ -0,0 +1,1885 @@
+	 * @param advancedControllableProperties is the list that store all controllable properties
+	 */
+	private void populateControlCapabilities(Map<String, String> stats, List<AdvancedControllableProperty> advancedControllableProperties) {
+		// Getting power status from device
+		PowerStatus powerStatus = getPowerStatus();
+		stats.put(Command.POWER.getName(), "");
+
+		assert powerStatus != null;
+		if (powerStatus.compareTo(PowerStatus.OFF) == 0) {
+			advancedControllableProperties.add(createSwitch(Command.POWER.getName(), 0, PowerStatus.OFF.getName(), PowerStatus.ON.getName()));
+		} else if (powerStatus.compareTo(PowerStatus.ON) == 0) {
+			advancedControllableProperties.add(createSwitch(Command.POWER.getName(), 1, PowerStatus.OFF.getName(), PowerStatus.ON.getName()));
+
+			// Zoom control
+			populateZoomControl(stats, advancedControllableProperties);
+
+			// Focus control
+			populateFocusControl(stats, advancedControllableProperties);
+
+			// AE control
+			populateAEControl(stats, advancedControllableProperties);
+
+			// WB control
+			populateWBControl(stats, advancedControllableProperties);
+
+			// Pan tilt control
+			populatePanTiltControl(stats, advancedControllableProperties);
+
+			// Preset control
+			populatePresetControl(stats, advancedControllableProperties);
+		}
+	}
+
+	//region Control device
+
+	/**
+	 * This method is used to control image process
+	 */
+	private void imageProcessControl(String value, String[] splitProperty) {
+		// RGain
+		if (splitProperty[1].equals(Command.RGAIN.getName() + RGainControl.UP.getName())) {
+			rGainControl(RGainControl.UP);
+			return;
+		} else if (splitProperty[1].equals(Command.RGAIN.getName() + RGainControl.DOWN.getName())) {
+			rGainControl(RGainControl.DOWN);
+			return;
+		}
+
+		// BGain
+		if (splitProperty[1].equals(Command.BGAIN.getName() + BGainControl.UP.getName())) {
+			bGainControl(BGainControl.UP);
+			return;
+		} else if (splitProperty[1].equals(Command.BGAIN.getName() + BGainControl.DOWN.getName())) {
+			bGainControl(BGainControl.DOWN);
+			return;
+		}
+
+		Command imageProcessCommand = Command.getByName(splitProperty[1]);
+		switch (imageProcessCommand) {
+			case WB_MODE:
+				wbModeControl(WBMode.getByName(value));
+				break;
+
+			case WB_ONE_PUSH_TRIGGER: {
+				wbOnePushTrigger();
+				break;
+			}
+			default: {
+				throw new IllegalStateException("Unexpected value: " + Arrays.toString(splitProperty));
+			}
+		}
+
+	}
+
+	/**
+	 * This method is used to control exposure
+	 */
+	private void exposureControl(String value, Command exposureCommand) {
+		switch (exposureCommand) {
+			case BACKLIGHT: {
+				if (value.equals(SWITCH_STATUS_ON)) {
+					backlightControl(BacklightStatus.ON);
+				} else if (value.equals(SWITCH_STATUS_OFF)) {
+					backlightControl(BacklightStatus.OFF);
+				}
+				break;
+			}
+			case AE_MODE: {
+				aeModeControl(AEMode.getByName(value));
+				break;
+			}
+			case EXP_COMP_DIRECT:
```

Comment:
each of the blocks without the _break_ statement will fall down to the next one, is it ok in this case? 
It is clear that there's no implementation for them, but my concern is that  SHUTTER_DIRECT case will be triggered as well if it is EXP_COMP_DIRECT, GAIN_LIMIT_DIRECT, GAIN_DIRECT or IRIS_DIRECT.

Reply:
ah I see, I will fix it, we need to break for each statement, thank you very much 😃